### PR TITLE
Make GrantConditionOnDeploy PausableConditional.

### PR DESCRIPTION
**Pull request by MustaphaTR**
#15154

### CODE REVIEW

**Comment by abcdefg30** [comment]
Looks like `CanUndeploy` was not removed entirely:
> OpenRA.Utility(1,1): Error: d2k|rules/vehicles.yaml:359 refers to a trait field `CanUndeploy` that does not exist on `GrantConditionOnDeploy`.

File: OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs [hunk 1]
```diff
@@ -20,7 +20,9 @@
 
 namespace OpenRA.Mods.Common.Traits
 {
-	public class GrantConditionOnDeployInfo : ITraitInfo
+	[Desc("Grants a condition when given a deploy order." +
```
Comment (GraionDilach) [inline OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs]: `when a deploy order was issued` IMO.

**Comment by GraionDilach** [comment]
Otherwise yes, that's what I meant with the desc. ✅

**Review by pchote — DISMISSED** [review]

**Comment by abcdefg30** [comment]
Needs a rebase, though.

**Comment by MustaphaTR** [comment]
Rebased.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/692b5a4ec3fcc0b86d5c8cc88214005e3aa1e4da

**Pull request merged**